### PR TITLE
clang-tidy: Apply modernize-pass-by-value.

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

**Comment by ekpyron** [comment]
I'm wondering whether we should really explicitly include ``<utility>`` basically everywhere... maybe we actually really should :-).

**Comment by leonardoalt** [comment]
> I'm wondering whether we should really explicitly include `<utility>` basically everywhere... maybe we actually really should :-).

I think we should.

**Comment by leonardoalt** [comment]
Needs rebase

File: libsolidity/analysis/ConstantEvaluator.h [hunk 1]
```diff
@@ -24,6 +24,10 @@
 
 #include <libsolidity/ast/ASTVisitor.h>
 
+
+#include <utility>
```
Comment (leonardoalt) [inline libsolidity/analysis/ConstantEvaluator.h:27]: Maybe a bit too many blanks?

File: libsolidity/ast/ASTVisitor.h [hunk 2]
```diff
@@ -22,9 +22,11 @@
 
 #pragma once
 
-#include <libsolidity/ast/AST.h>
 #include <functional>
+#include <libsolidity/ast/AST.h>
```
Comment (leonardoalt) [inline libsolidity/ast/ASTVisitor.h:25]: This was correct before (specific first, general after)

File: libsolidity/ast/AsmJsonImporter.h [hunk 3]
```diff
@@ -26,6 +26,10 @@
 #include <liblangutil/SourceLocation.h>
 #include <libyul/AsmDataForward.h>
 
+
+#include <utility>
```
Comment (leonardoalt) [inline libsolidity/ast/AsmJsonImporter.h:29]: Too many blanks

File: libsolidity/ast/Types.cpp [hunk 4]
```diff
@@ -43,6 +43,8 @@
 #include <boost/range/algorithm/copy.hpp>
 
 #include <limits>
+#include <utility>
+
```
Comment (leonardoalt) [inline libsolidity/ast/Types.cpp:47]: Extra blank

File: libsolidity/interface/CompilerStack.cpp [hunk 5]
```diff
@@ -66,6 +66,8 @@
 #include <json/json.h>
 
 #include <boost/algorithm/string.hpp>
+#include <utility>
+
```
Comment (leonardoalt) [inline libsolidity/interface/CompilerStack.cpp:72]: Extra blank

File: libsolutil/Common.h [hunk 6]
```diff
@@ -47,6 +47,8 @@
 #include <boost/multiprecision/cpp_int.hpp>
 
 #include <map>
+#include <utility>
+
```
Comment (leonardoalt) [inline libsolutil/Common.h]: Extra blank

File: libyul/AsmAnalysis.h [hunk 7]
```diff
@@ -34,6 +34,8 @@
 #include <list>
 #include <memory>
 #include <optional>
+#include <utility>
+
```
Comment (leonardoalt) [inline libyul/AsmAnalysis.h:38]: Extra blank

File: libyul/backends/evm/EVMCodeTransform.cpp [hunk 8]
```diff
@@ -29,6 +29,8 @@
 
 #include <boost/range/adaptor/reversed.hpp>
 
+#include <utility>
+
```
Comment (leonardoalt) [inline libyul/backends/evm/EVMCodeTransform.cpp]: Extra blank

File: libyul/optimiser/FullInliner.h [hunk 9]
```diff
@@ -31,6 +31,8 @@
 
 #include <optional>
 #include <set>
+#include <utility>
+
```
Comment (leonardoalt) [inline libyul/optimiser/FullInliner.h:35]: Extra blank

File: test/tools/isoltest.cpp [hunk 10]
```diff
@@ -31,10 +31,12 @@
 #include <boost/program_options.hpp>
 
 #include <cstdlib>
-#include <iostream>
 #include <fstream>
+#include <iostream>
 #include <queue>
 #include <regex>
+#include <utility>
+
```
Comment (leonardoalt) [inline test/tools/isoltest.cpp:39]: Extra blank

File: tools/solidityUpgrade/UpgradeChange.h [hunk 11]
```diff
@@ -21,6 +21,8 @@
 #include <liblangutil/SourceLocation.h>
 
 #include <algorithm>
+#include <utility>
+
```
Comment (leonardoalt) [inline tools/solidityUpgrade/UpgradeChange.h:25]: Extra blank

File: libsolidity/ast/AST.h [hunk 12]
```diff
@@ -38,6 +38,8 @@
 #include <memory>
 #include <optional>
 #include <string>
+#include <utility>
+
```
Comment (leonardoalt) [inline libsolidity/ast/AST.h]: Extra blank

**Review by leonardoalt — COMMENTED** [review]
Looks good! Just a bunch of small comments

**Review by leonardoalt — DISMISSED** [review]

**Comment by chriseth** [comment]
Needs rebase.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**